src/Elasticsearch/ObjectTrait.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace Drupal\elasticsearch_helper\Elasticsearch;
+
+use Drupal\Component\Utility\NestedArray;
+
+/**
+ * Provides helper methods to object definition classes.
+ */
+trait ObjectTrait {
+
+  /**
+   * @var array
+   */
+  protected $options = [];
+
+  /**
+   * Returns new instance.
+   *
+   * @return static
+   */
+  public static function create() {
+    $instance = new static();
+
+    return $instance;
+  }
+
+  /**
+   * Adds an option.
+   *
+   * @param $option
+   * @param $value
+   *
+   * @throws \InvalidArgumentException
+   *
+   * @return $this
+   */
+  public function addOption($option, $value) {
+    $options = [$option => $value];
+    $this->validateOptions($options);
```

Comment:
`$this->addOptions($options)` could just be called here instead.

Reply:
Good point!